### PR TITLE
[MI-2440] Resolved TODO comments and refactored webapp code

### PR DESCRIPTION
**Pull request by ayusht2810**
1. Resolved TODO comments.
2. Refactored webapp code by adding new hooks and updating the existing ones and updating components according to them.
3. Added some new enhancements and bug fixes.

### CODE REVIEW

File: webapp/src/containers/shareRecords/index.tsx [hunk 1]
```diff
@@ -76,22 +71,16 @@ const ShareRecords = () => {
...
-            setShowResultPanel(true);
-        }
-
-        setShowModalLoader(isLoading);
-    }, [getShareRecordState().isLoading, getShareRecordState().isError, getShareRecordState().isSuccess]);
+    useApiRequestCompletionState({
+        serviceName: Constants.pluginApiServiceConfigs.shareRecord.apiServiceName,
+        payload: shareRecordPayload,
+        handleSuccess: () => setShowResultPanel(true),
+        handleError: (error) => setApiError(error),
```
Comment (manojmalik20) [inline webapp/src/containers/shareRecords/index.tsx:82]:
```suggestion
        handleError: setApiError,
```

File: webapp/src/containers/updateState/index.tsx [hunk 2]
```diff
@@ -72,41 +72,35 @@ const UpdateState = () => {
...
+        setApiError(error);
+        setShowResultPanel(true);
+    };
 
-        if (isSuccess) {
+    useApiRequestCompletionState({
+        serviceName: Constants.pluginApiServiceConfigs.getStates.apiServiceName,
+        payload: getStatesParams,
+        handleSuccess: () => setApiError(null),
+        handleError: (error) => handleError(error),
```
Comment (manojmalik20) [inline webapp/src/containers/updateState/index.tsx:92]: Same. Fix this everywhere applicable.

File: webapp/src/hooks/useApiRequestCompletionState.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    useEffect(() => {
+        if (
+            getApiRequestCompletionState(pluginState).requests.includes(serviceName) &&
+            getApiState(serviceName, payload)
+        ) {
+            const {isError, isSuccess, isUninitialized, error} = getApiState(serviceName, payload);
+            const apiErr = (error as FetchBaseQueryError)?.data as APIError | undefined;
+            if (isSuccess && !isError) {
+                // eslint-disable-next-line no-unused-expressions
+                handleSuccess?.();
```
Comment (manojmalik20) [inline webapp/src/hooks/useApiRequestCompletionState.ts:31]:
```suggestion
            if (isSuccess && !isError && handleSuccess) {
                handleSuccess();
```

File: webapp/src/hooks/useApiRequestCompletionState.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            }
+
+            if (!isSuccess && isError && apiErr) {
+                // eslint-disable-next-line no-unused-expressions
+                handleError?.(apiErr);
```
Comment (manojmalik20) [inline webapp/src/hooks/useApiRequestCompletionState.ts:36]: Same.

File: webapp/src/plugin_constants/index.ts [hunk 5]
```diff
@@ -52,6 +52,16 @@ const RequestButtonRedirectText = 'You will be redirected to ServiceNow to compl
 const DefaultPerPageParam = 10;
 const DebounceFunctionTimeLimit = 500;
 
+export enum ModalId {
```
Comment (manojmalik20) [inline webapp/src/plugin_constants/index.ts:55]:
```suggestion
export enum ModalIds {
```

File: webapp/src/reducers/connectedState/index.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import {createSlice, PayloadAction} from '@reduxjs/toolkit';
 
 const initialState: ConnectedState = {
-    connected: false,
+    connected: true,
```
Comment (manojmalik20) [inline webapp/src/reducers/connectedState/index.ts:4]: Why this change?

File: server/plugin/command.go [hunk 7]
```diff
@@ -240,7 +240,7 @@ func (p *Plugin) handleSearchAndShare(_ *plugin.Context, args *model.CommandArgs
 	return ""
 }
 
-func (p *Plugin) handleListSubscriptions(_ *plugin.Context, args *model.CommandArgs, params []string, client Client, isSysAdmin bool) string {
+func (p *Plugin) handleListSubscriptions(args *model.CommandArgs, params []string, client Client, isSysAdmin bool) string {
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/plugin/command.go:243]: please check this
https://github.com/mattermost/mattermost-plugin-servicenow/pull/125#discussion_r1031669358

[the remark this answers is not in the document]
Comment (manojmalik20) [reply server/plugin/command.go:243]: Then you should remove context from all the functions as it is not being used in any function.

**Review by manojmalik20 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]
